src/csharp/Grpc.Core/Internal/PlatformApis.cs

```diff
@@ -63,7 +64,25 @@ static PlatformApis()
             isNetCore = false;
 #endif
             isMono = Type.GetType("Mono.Runtime") != null;
-            isUnity = Type.GetType(UnityEngineApplicationClassName) != null;
+
+            // Unity
+            var unityApplicationClass = Type.GetType(UnityEngineApplicationClassName);
```

Comment:
is this enough of a check?

Reply:
I think so, there's no official way so we just chose an approach that seemed most reasonable (also the general approach stays the same in this PR, so not really an issue we need to solve right now).